### ISSUE TITLE
debugger controls disappear after console action

### ISSUE THREAD

**Issue by aronatkins** [rstudio/rstudio#10045]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2022.01.0 Build 183
    OS Version      : macOS Big Sur 11.6
    R Version       : 3.6.3

### Steps to reproduce the problem

Opened the `rsconnect` project and added a breakpoint to the very start of `rsconnect::deploySite` (in `R/deploySite.R`).
Called this function.

```
rsconnect::deploySite(siteDir = ".")
```

Click the "next" debugger button to run the `oldwd` line.
From the console, print the value of `oldwd`.

```
Browse[2]> oldwd
[1] "/Users/aron/dev/rstudio/standalone/bookdown-again"
```

Click the "next" debugger button.
The debugger controls disappear!

### Describe the problem in detail

The debugger console still appears to be active (typing `n` advances the code), but the debugger controls are gone and the source window no longer shows the current line.

### Describe the behavior you expected

The debugger controls should remain and the source viewer indicates the current line.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by aronatkins** [comment]
The debugger controls return and the source window current-line indication may eventually return given enough interaction with the debugger in the console. I was manually entering "n" and eventually saw things return to their expected state; this may require the "next" command to enter a separate code block (a `for { .. }` block in my case).

**Comment by aronatkins** [comment]
When exiting that inner `for { ... }` block, the debugger controls are lost again.

**Comment by mikebessuille** [comment]
@Ron is this a regression?

**Comment by ronblum** [comment]
[Following some help from @aronatkins ]

This is a regression—in Ghost Orchid patch 1, 2021.09.1+371. It seems to work fine in build 351. I'll try some more to see if I can reproduce it in 351, though.

**Comment by jmcphers** [comment]
If it is indeed a regression in +371, it could be a regression from https://github.com/rstudio/rstudio/pull/9922.

**Comment by aronatkins** [comment]
Alternate steps that do not require an R package:

```r
# Create a new project with this code in `important.R`.
important <- function(half = 21) {
  # First breakpoint here; run "n", then type "answer" to inspect
  # its value in the console, then run "n" again. After that second
  # "n", the debugger controls disappear. Run "c".
  answer <- 2 * 3 * 7
  vals <- seq(from = 1, to = 1)
  doubles <- lapply(vals, function(val) {
    # Second breakpoint here; the controls return.
    # Run "c".
    double <- val * 2
    return(double)
  })
  # Third breakpoint here. Run "n". The controls disappear again.
  # Run "c" to finish the function.
  cat("the answer is ", answer, "\n", sep = "")
  invisible(doubles)
}
```

Source this file, add breakpoints as suggested, then run `important()` from the R console.

**Comment by ronblum** [comment]
Confirmed
- Works correctly in 2021.09.0+351
- UI disappears in 2021.09+1+371

**Comment by R-Hannibal** [comment]
@jmcphers - The issue you tagged (https://github.com/rstudio/rstudio/pull/9922) as possible source of this regression is a PT PR and PT issue. Shouldn't the GO patch not include PT fixes, or am I taking crazy pills again?

**Comment by jmcphers** [comment]
We added that bugfix to the GO branch because of a whole bunch of reports of the original issue plus no known regressions or issues. If that bugfix was indeed the cause we will want to back it out of GO.

**Comment by mikebessuille** [comment]
Yes we should back it out of the GO patch... and then we should fix it correctly in PT (using this issue to fix the root of the problem).

**Comment by R-Hannibal** [comment]
This is in fact looking like it's caused by the #9922 fix - 

Re-tested this on `2021.12.0-daily+91` (PT build prior to the #9922 check-in) and the debug controls do not disappear.

**Comment by R-Hannibal** [comment]
This issue no longer occurs in the most recent GO patch build `2021.09.1+372`. Debugger controls remain visible in the console pane until debugging is completed.

**Comment by jmcphers** [comment]
"Fixed" in Ghost Orchid by backing out #9922 on that branch; needs to be fixed in PT more comprehensively.

**Comment by ronblum** [comment]
Verified in
- RStudio Desktop RStudio 2022.02.0-daily+324 on MacOS 12.1
- RStudio Server 2022.02.0-daily+318 on Ubuntu 20.04.2 via Safari on MacOS 12.1

Note that I limited this test to the reprexes from above, and confirmed that #9918 is unaffected.. A request for a more comprehensive test of the debugger is available at https://github.com/rstudio/rstudio-ide-automation/issues/497, which references these two issues.
